olp-cpp-sdk-core/include/olp/core/thread/ExecutionContext.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2022 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <utility>
+
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/CancellationContext.h>
+
+namespace olp {
+namespace thread {
+
+/// Handles the cancellation and final mechanisms.
+class CORE_API ExecutionContext final {
+  using CancelFuncType = std::function<void()>;
+  using ExecuteFuncType = std::function<client::CancellationToken()>;
+  using FailedCallback = std::function<void(client::ApiError)>;
+
+ public:
+  /// A default contructor, initializes the `ExecutionContextImpl` instance.
+  ExecutionContext();
+
+  /**
+   * @brief Sets the error that is returned in the `Finally`
+   * method of `Continuation`.
+   *
+   * Method is intended to be used in `Then` methods of
+   * `TaskContinuation`/`Continuation`. It immediately finishes an
+   * execution of task continuation and provides an error in
+   * the `Finally` method of `Continuation`.
```

Comment:
Theoretically the `ExecutionContext` can be used for anything, it does not provide exclusive rights to  `Continuation` only, so you might want to adapt this.

Reply:
done